src/faction_camp.cpp

```diff
@@ -0,0 +1,3443 @@
+    const tripoint omt_pos = p.global_omt_location();
+    tripoint forest = om_target_tile( omt_pos, 1, 50, log_sources );
+    if( forest != tripoint( -999, -999, -999 ) ) {
+        int tree_est = om_cutdown_trees( forest, .50, false, false );
+        int tree_young_est = om_harvest_ter( p, forest, ter_id( "t_tree_young" ), .50, false );
+        int dist = rl_dist( forest.x, forest.y, omt_pos.x, omt_pos.y );
+        //Very roughly what the player does + 6 hours for prep, clean up, breaks
+        time_duration chop_time = 6_hours + 1_hours * tree_est + 7_minutes * tree_young_est;
+        //Generous to believe the NPC can move ~ 2 logs or ~8 heavy sticks (3 per young tree?) per trip, each way is 1 trip
+        // 20 young trees => ~60 sticks which can be carried 8 at a time, so 8 round trips or 16 trips total
+        //This all needs to be in an om_carry_weight_over_distance function eventually...
+        int trips = tree_est + tree_young_est * 3  / 4;
+        //Alwasy have to come back so no odd number of trips
+        trips += trips & 1 ? 1 : 0;
+        time_duration travel_time = companion_travel_time_calc( forest, omt_pos, 0_minutes, trips );
+        time_duration work_time = travel_time + chop_time;
+        int need_food = time_to_food( work_time );
+        if( !query_yn( _( "Trip Estimate:\n%s" ), camp_trip_description( work_time, chop_time, travel_time,
+                       dist, trips, need_food ) ) ) {
+            return;
+        } else if( camp_food_supply() < need_food ) {
+            popup( _( "You don't have enough food stored to feed your companion." ) );
+            return;
+        }
+        g->draw_ter();
+        //wrefresh( g->w_terrain );
+        npc *comp = individual_mission( p, _( "departs to cut logs..." ), "_faction_camp_cut_log", false, {},
+                                        "fabrication", 2 );
+        if( comp != nullptr ) {
+            units::mass carry_m = comp->weight_capacity() - comp->weight_carried();
+            // everyone gets at least a makeshift sling of storage
+            units::volume carry_v = comp->volume_capacity() - comp->volume_carried() + item(
+                                        itype_id( "makeshift_sling" ) ).get_storage();
+            om_cutdown_trees( forest, .50 );
+            om_harvest_ter( *comp, forest, ter_id( "t_tree_young" ), .50, true );
+            std::pair<units::mass, units::volume> harvest = om_harvest_itm( comp, forest, .95, true );
+            // recalculate trips based on actual load and capacity
+            trips = om_carry_weight_to_trips( harvest.first, harvest.second, carry_m, carry_v );
+            travel_time = companion_travel_time_calc( forest, omt_pos, 0_minutes, trips );
+            work_time = travel_time + chop_time;
+            comp->companion_mission_time_ret = calendar::turn + work_time;
+            //If we cleared a forest...
+            tree_est = om_cutdown_trees( forest, .50, false, false );
+          
```

Comment:
Parenthesis on the right are not necessary.

Reply:
Fixed, thanks.